private/buf/cmd/buf/command/beta/convert/convert.go

```diff
@@ -164,8 +167,30 @@ func run(
 		false, // externalDirOrFilePathsAllowNotExist
 		false, // excludeSourceCodeInfo
 	)
-	if err != nil {
-		return err
+	var resolveWellKnownType bool
+	// only resolve wkts if input was not set.
+	if container.NumArgs() == 0 {
+		if inputErr != nil {
+			resolveWellKnownType = true
+		}
+		if image != nil {
+			_, filterErr := bufimageutil.ImageFilteredByTypes(image, flags.Type)
+			if errors.Is(filterErr, bufimageutil.ErrImageFilterTypeNotFound) {
+				resolveWellKnownType = true
+			}
+		}
```

Comment:
This seems odd. So if they provide inputs to identify the source of the well-known type, but they give the wrong inputs, we just ignore them and use the well-known type data? Seems like we should only fall back to the well-known type data when (1) no input files are named or (2) they name a file that is _in_ the well-known type data and _not_ found locally. Wouldn't both of those conditions result in `inputErr != nil` above?